tapchannel/aux_invoice_manager.go

```diff
@@ -197,6 +198,28 @@ func (s *AuxInvoiceManager) handleInvoiceAccept(_ context.Context,
 	// Convert the total asset amount to milli-satoshis using the price from
 	// the accepted quote.
 	rfqID := htlc.RfqID.ValOpt().UnsafeFromSome()
+
+	// Retrieve the asset identifier from the RFQ quote.
+	identifier, err := s.identifierFromQuote(rfqID)
+	if err != nil {
+		return nil, fmt.Errorf("could not extract assetID from "+
+			"quote: %v", err)
+	}
+
+	// Check for each of the asset balances of the HTLC that the identifier
+	// matches that of the RFQ quote.
+	for _, v := range htlc.Balances() {
+		err := checkHTLCAssetSpecifierMatch(v, identifier)
+		if err != nil {
+			log.Errorf("asset specifier mismatch, cancelling "+
+				"htlc set: %v", err)
+
+			resp.CancelSet = true
+
+			return resp, nil
+		}
+	}
```

Comment:
I think we should consider rolling this functionality into a method called:
```
func (s *AuxInvoiceManager) validateHTLC(htlc rfqmsg.Htlc) error
```

* That method could extract `rfqID` from the given `htlc`.
* Call `s.identifierFromQuote` to get the associated asset specifier.
* Validate HTLC balances.

And then we can drop `checkHTLCAssetSpecifierMatch` which is a less general HTLC verifier.

Reply:
yeah, agree with this simplification

I think @guggero 's recommendation already leaned towards this direction, so we might as well fully abstract this away (since it's gonna change a lot soon-ish anyway)